cpp/src/glm/ridge_mg.cu

```diff
@@ -116,23 +113,23 @@ void ridgeEig(raft::handle_t& handle,
   std::vector<Matrix::RankSizePair*> partsToRanks = ADesc.blocksOwnedBy(rank);
   size_t total_size                               = 0;
 
-  for (int i = 0; i < partsToRanks.size(); i++) {
+  for (std::size_t i = 0; i < partsToRanks.size(); i++) {
```

Comment:
Is it necessary to prefix `std::` all the time?

Reply:
Not strictly, but this is C++, and `size_t` is defined in the C++ standard library in the namespace `std` in `<cstddef>`.  Technically it is also defined in the global namespec in `stddef.h`. But I think not being explicit could lead to unforseen issues.

Perhaps more importantly for me is that clangd, which I use for "intellisense" in my IDE, puts a red squiggle under `size_t` but not under `std::size_t`.  I usually take clangd warnings to be as important as gcc warnings for avoiding potential issues.

https://stackoverflow.com/a/8507891/749748

Reply:
I use clangd too, but with language server and emacs, so far it works fine.  Just a quick question, I don't have a strong preference for the choice of the prefix.

Reply:
I would urge us to stick with `std::size_t` wherever we need a `size_t` unless we are writing an interface to C code. `std::size_t` clearly indicates a C++ features, and `size_t` clearly indicates that we are in the realm of C. Furthermore `stddef.h` has "been identified as a candidate for removal from future revisions" of the C++ standard and `size_t` is not guaranteed to be defined anywhere else.